### ISSUE TITLE
Investigate RDF -> LPG generalized collapsing process

### ISSUE THREAD

**Issue by zwelz3** [jupyrdf/ipyradiant#50]
Continuation of #42. 

Previous efforts have identified a basic process for collapsing a set of predicates onto their subject nodes in order to transform an RDF graph into an LPG graph. This ticket expands upon that process and attempts to identify relevant patterns for higher-order collapsing (e.g. more predicates). 

Scope (completion conditions):
* [x] Survey of open-source RDF->LPG capabilities
* [x] Initial implementation in ipyradiant
  * [x] Basic query steps
    * [x] NodeIRIs
    * [x] NodeTypes
    * [x] NodeProperties
    * [x] RelationTypes
    * [x] ReifiedRelations
    * [x] RelationProperties
    * Singletons *(skipping for now)*
  * Examples of common patterns for each query *(skipping for now)*
  * [x] Literal converters
* [x] Better understanding of common collapsing operations (multi-edges, disparate data, etc.)
* [x] Considerations on ways to visualize what is happening in the collapsed graph
* [x] Documentation and notebook clean-up

**Comment by zwelz3** [comment]
~~TODO~~ discussion about collapsing multiple predicate objects (e.g. `ns:has "Thing1", ns:has "Thing2"`)

After a brief discussion, it appears that we will write these to a list

```
Node1 ns:pred1 """Thing1"""
Node1 ns:pred1 """Thing2"""
```
to
```
Node1 : {
  "ns:pred1": [
    """Thing1""",
    """Thing2"""
  ]
}
```

Addressed in [this commit](https://github.com/jupyrdf/ipyradiant/pull/58/commits/986dc4aed2b5620d9eda55698ecb924fabdeaba6#diff-2de7641580c20c2969daad20ee27b3ea0e6c295acadc17a829ceec382657fcc7R40)

**Comment by zwelz3** [comment]
~~TODO discussion about classes that store `uri`, `label`, `namespace`, etc.~~
deferred to a future ticket

**Comment by zwelz3** [comment]
> * [ ] Considerations on ways to visualize what is happening in the collapsed graph

Need a PR to bring in ipyelk, but we have to wait on lab 3.0
#66
